File: leetcode/2sum.cpp
//Given an array of integers, find two numbers such that they add up to a specific target number.
//a + b = target, target - b = a;
//index 1 
vector <int> twoSum(vector<int>& nums, int target){
	unordered_map<int, int>hash;
	vector<int> result;
	
	for (int i = 0; i < numbers.size(); i++){
		int numberToFind = target - numbers[i];

		if (hash.find(numberToFind) != hash.end()){ //find returns hash.end if nothing found
			result.push_back(hash[numberToFind] + 1); //represents 
			result.push_back(i+1);
			return result;
		}
		
		hash[numbers[i]] = i;
	
	
	}


}

File: leetcode/2sum.js
//Using a map and returning indices

function twoSum_(nums, target) { 

}

//48ms 99%
function twoSum(nums, target) {
  let map = new Map();
  for (let i = 0; i < nums.length; i++) {
    const complement = target - nums[i];
    if (map.has(complement)) {
      return [map.get(complement), i];
    } else {
      map.set(nums[i], i);
    }
  }
  return [];
}


